app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepo.kt

```diff
@@ -0,0 +1,152 @@
+            if (path.fileName.extension == NOTE_EXT) {
+                try {
+                    val data = StringBuilder()
+                    path.forEachLine {
+                        data.appendLine(it)
+                    }
+                    val size = path.fileSize()
+                    val id = computeId(size, path)
+                    val meta = ResourceMeta(
+                        id,
+                        path.fileName.name,
+                        path.extension,
+                        path.getLastModifiedTime(),
+                        size
+                    )
+                    val titles = propertiesStorage.getProperties(id).titles
+                    val content = TextNote.Content(titles.elementAt(0), data.toString())
+                    val note = TextNote(content, meta)
+                    notes.add(note)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
+            }
+        }
+        Log.d("text-repo", "${notes.size} text note resources found")
+        _textNotes.value = notes
+    }
+
+    private suspend fun write(note: TextNote) = withContext(Dispatchers.IO) {
+        val path = root.resolve("${DUMMY_FILENAME}.${NOTE_EXT}")
+        if (!path.exists()) {
+            try {
+                val lines = note.content.data.split(NEWLINE)
+                path.writeLines(lines)
+                val size = path.fileSize()
+                val id = computeId(size, path)
+                val properties = Properties(setOf(note.content.title), setOf())
+
+                Log.d("text-repo", "initial resource name ${path.name}")
+
+                propertiesStorage.setProperties(id, properties)
+                propertiesStorage.persist()
+
+                val newPath = root.resolve("$id.$NOTE_EXT")
+                if (!newPath.exists()) {
+                    if (path.toFile().renameTo(newPath.toFile())) {
+                        note.meta = ResourceMeta(
+                            id,
+                            newPath.fileName.name,
+                            newPath.extension,
+                            newPath.getLastModifiedTime(),
+                            size
+                        )
+                        Log.d("notes-repo", "resource renamed to ${newPath.name} successfully")
+                    } else delete(path)
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+            }
+        }
+        this.coroutineContext.job
+    }
+
+    private fun delete(path: Path) {
+        path.deleteIfExists()
+    }
+
+    private fun add(note: TextNote) {
+        val notes = textNotes.value.toMutableList()
+        notes.add(note)
+        _textNotes.value = notes
+    }
+
+    private fun remove(note: TextNote) {
+        val notes = textNotes.value.toMutableList()
+        notes.remove(note)
+        _textNotes.value = notes
+    }
+
+    companion object {
```

Comment:
Let's move these constants to be at top of the file, and below class declaration. It is recommended that we should use top-declaration
https://stackoverflow.com/questions/44038721/constants-in-kotlin-whats-a-recommended-way-to-create-them

```kotlin
private const val NEWLINE = "\n"
private const val NOTE_EXT = "note"
private const val DUMMY_FILENAME = "Note"

class TextNotesRepo {
    ...
}